src/EnergyPlus/UnitarySystem.cc

```diff
@@ -17856,5 +17891,147 @@ namespace UnitarySystems {
+                        mixedTempAtMinOA =
+                            (outdoorAirTemp * outdoorAirController.MinOA + state.dataLoopNodes->Node(outdoorAirController.RetNode).Temp *
+                                                                               (highSpeedEconMassFlowRate - outdoorAirController.MinOA)) /
+                            highSpeedEconMassFlowRate;
+                    } else {
+                        mixedTempAtMinOA = outdoorAirTemp;
+                    }
+                    econClgOutputMinOA = cpAir * highSpeedEconMassFlowRate * (zoneTemp - (mixedTempAtMinOA + highSpeedFanDT));
+                    if (econClgOutputMinOA < std::abs(zoneLoad)) {
+                        highSpeedEconRuntime = 1.0;
+                    } else {
+                        // if running at this speed would "overcool", we run partly at the lower speed and partly at this speed
+                        lowSpeedEconMassFlowRate = this->m_CoolMassFlowRate[clgSpd - 1];
+                        lowSpeedFanDT = this->getFanDeltaTemp(state,
+                                                              firstHVACIteration,
+                                                              lowSpeedEconMassFlowRate,
+                                                              lowSpeedEconMassFlowRate / this->m_CoolMassFlowRate[this->m_NumOfSpeedCooling]);
+                        this->m_LowSpeedEconOutput = cpAir * lowSpeedEconMassFlowRate * (zoneTemp - (outdoorAirTemp + lowSpeedFanDT));
+                        // determine this speed's runtime
+                        highSpeedEconRuntime = (std::abs(zoneLoad) - this->m_LowSpeedEconOutput) / (econClgOutput - this->m_LowSpeedEconOutput);
+                    }
+                } else {
+                    highSpeedEconRuntime = 1.0;
+                }
+                // set economizer air flow "speed"
+                this->m_EconoSpeedNum = clgSpd;
+                // set economizer PLR, a.k.a the system fan part load ratio, and runtime at each speed
+                this->m_EconoPartLoadRatio = highSpeedEconRuntime;
+                this->m_LowSpeedEconRuntime = 1 - highSpeedEconRuntime;
+                break;
+            }
+        }
+    }
+
+    void UnitarySys::calcMixedTempAirSPforEconomizerStagingOperation(EnergyPlusData &state,
+                                                                     int const airLoopNum,
+                                                                     bool const firstHVACIteration,
+                                                                     Real64 const zoneLoad)
+    {
+        auto outdoorAirController = state.dataMixedAir->OAController(this->OAControllerIndex);
+        using Psychrometrics::PsyCpAirFnW;
+        Real64 cpAir = PsyCpAirFnW(state.dataLoopNodes->Node(outdoorAirController.InletNode).HumRat);
+        Real64 zoneTemp = state.dataLoopNodes->Node(state.dataZoneEquip->ZoneEquipConfig(this->ControlZoneNum).ZoneNode).Temp;
```

Comment:
@lymereJ if the UnitarySystem is set up for Setpoint control, this->ControlZoneNum will be 0. Can you quickly test that to make sure there is no crash here, or check for Load control early on so this doesn't crash?

Reply:
We do check for load control early on but we just use the outcome to display a warning the system is controlled by setpoint, so I think it will crash. I can add a check for type of control in the if statement below so we don't perform economizer staging calculations if the system is setpoint controlled.
```
this->m_EconoSpeedNum = 0;
if (this->OAControllerEconomizerStagingType == DataHVACGlobals::EconomizerStagingType::EconomizerFirst) {
    manageEconomizerStagingOperation(state, AirLoopNum, FirstHVACIteration, ZoneLoad);
}
```
Would that work for you?

Reply:
I checked and I don't think we need to do anything here. `manageEconomizerStagingOperation` is only called from `controlUnitarySystemOutput` which is only used for load based systems. So if a model uses the new `Controller:OutdoorAir` input, it won't change anything to the simulation. Moreover, the PR added a warning message to let the user know that that if that input is selected in that context it won't do anything.